AppTaskQueue/appscale/taskqueue/queue.py

```diff
@@ -235,7 +235,538 @@ def __repr__(self):
+          'retry_count': 0,
+          'tag': getattr(task, 'tag', None)
+        }
+      )
+
+      self.pg_connection.commit()
+      logger.debug('Added task: {}'.format(task))
+
+    except psycopg2.IntegrityError as err:
+      name_taken_msg = 'Task name already taken: {}'.format(task.id)
+      logger.warning('{msg}. Rolling back transaction({err})'
+                     .format(msg=name_taken_msg, err=err))
+      self.pg_connection.rollback()
+      raise InvalidTaskInfo(name_taken_msg)
+    except Exception as err:
+      logger.error('Rolling back transaction ({err})'.format(err=err))
+      self.pg_connection.rollback()
+      raise
+
+  def get_task(self, task, omit_payload=False):
+    """ Gets a task from the queue.
+
+    Args:
+      task: A Task object.
+      omit_payload: A boolean indicating that the payload should not be
+        fetched.
+    Returns:
+      A task object or None.
+    """
+    if omit_payload:
+      columns = ['task_name', 'time_enqueued',
+                 'lease_expires', 'retry_count', 'tag']
+    else:
+      columns = ['payload', 'task_name', 'time_enqueued',
+                 'lease_expires', 'retry_count', 'tag']
+    try:
+      self.pg_cursor.execute(
+        'SELECT {columns} FROM "{tasks_table}" '
+        'WHERE task_name = %(task_name)s'
+        .format(columns=', '.join(columns),
+                tasks_table=self.tasks_table_name),
+        vars={
+          'task_name': task.id,
+        }
+      )
+      row = self.pg_cursor.fetchone()
+      self.pg_connection.commit()
+    except Exception as err:
+      logger.error('Rolling back transaction ({err})'.format(err=err))
+      self.pg_connection.rollback()
+      raise
+
+    if not row:
+      return None
+    return self._task_from_row(columns, row, id=task.id)
+
+  def delete_task(self, task):
+    """ Deletes a task from the queue.
+
+    Args:
+      task: A Task object.
+    """
+    try:
+      self.pg_cursor.execute(
+        'DELETE FROM "{tasks_table}" '
+        'WHERE "{tasks_table}".task_name = %(task_name)s'
+        .format(tasks_table=self.tasks_table_name),
+        vars={
+          'task_name': task.id,
+        }
+      )
+      self.pg_connection.commit()
+    except Exception as err:
+      logger.error('Rolling back transaction ({err})'.format(err=err))
+      self.pg_connection.rollback()
+      raise
+
+  def update_lease(self, task, new_lease_seconds):
+    """ Updates the duration of a task lease.
+
+    Args:
+      task: A Task object.
+      new_lease_seconds: An integer specifying when to set the new ETA. It
+        represents the number of seconds from now.
+    Returns:
+      A Task object.
+    """
+    new_eta = current_time_ms() + datetime.timedelta(seconds=new_lease_seconds)
+    try:
+      self.pg_cursor.execute(
+        'UPDATE "{tasks_table}" '
+        'SET lease_expires = %(new_eta)s '
+        'WHERE task_name = %(task_name)s '
+        '  AND lease_expires > %(current_time)s '
```

Comment:
Any time the current time is need for a db operation, the native postgres function should be used.